components/gitpod-db/src/typeorm/migration/1655988518555-WorkspaceInstanceUsageAttributionId.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists } from "./helper/helper";
+
+const TABLE_NAME = "d_b_workspace_instance";
+const COLUMN_NAME = "usageAttributionId";
+
+export class WorkspaceInstanceUsageAttributionId1655988518555 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, TABLE_NAME, COLUMN_NAME))) {
+            await queryRunner.query(
+                `ALTER TABLE ${TABLE_NAME} ADD COLUMN ${COLUMN_NAME} varchar(255) NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE`,
```

Comment:
> varchar(255)

What about making it smaller in max size? Mentioning this because the bigger the key, the bigger the index and the more costly operations on it will be. :watch:

Reply:
Good catch! But `varchar`'s exact size depends on the actual content, right? And with one address byte, you can go up to 255 characters. So, I think there is no advantage in using `varchar(50)` over `varchar(255)`, other than preventing ourselves from using larger sizes one day (either by accident or because we need to).

From the [Drupal docs](https://www.drupal.org/docs/7/api/schema-api/data-types/varchar)

> **Storage Information :**
> L + 1 bytes if column values require 0 − 255 bytes, L + 2 bytes if values may require more than 255 bytes where L is L is the byte length of the string that you are storing and not the maximum size that you have set.

Reply:
That's true for the column, but not for the indexex: It has to assume the max size for a given column. And that directly affects query times, especially if the index becomes too big for RAM (as we see for this table already).

If we want to later `GROUP BY` on this, an index is required.

Reply:
Ah, that's an excellent point, thank you @geropl!

So, should I:
- change the type to `char(41)` (exactly enough for `team:` or `user:` prefix followed by UUID)?
- or make it slightly bigger? (e.g. `char(50)`?)
- and add an index to it in this migration?

Reply:
I'm fine with sth like `char(50)` or `char(60)` to anticipate slighty longer prefixes. :+1: 

> and add an index to it in this migration?

Let's leave that for a follow-up. :lotus_position:

Reply:
> I'm fine with sth like `char(50)` or `char(60)` to anticipate slighty longer prefixes. 👍

How about `varchar(60)`? 😁

> Let's leave that for a follow-up. 🧘

I think it's just a couple more characters in the migration, right? 😇 Maybe I can add it now.